src/endpoint/s3/s3_utils.js

```diff
@@ -131,7 +131,22 @@ function get_request_xattr(req) {
 }
 
 function set_response_xattr(res, xattr) {
-    _.each(xattr, (val, key) => {
+    let size_for_md_left = config.S3_MD_SIZE_LIMIT;
+    const sliced_xattr = {};
+    const sorted_xattr = Object.keys(xattr).sort().reduce((obj, key) => {
+        obj[key] = xattr[key];
+        return obj;
+    }, {});
+    for (const [key, value] of Object.entries(sorted_xattr)) {
+        const property_size = Buffer.byteLength(key, 'utf8') + Buffer.byteLength(value, 'utf8');
+        if (property_size > size_for_md_left) {
+            res.setHeader('x-amz-missing-meta', Object.keys(sorted_xattr).length - Object.keys(sliced_xattr).length);
```

Comment:
Better maintain this count instead of regenerate whole lists of keys just to get the length.